tests/Collection/CrudSpecFunctionalTest.php

```diff
@@ -334,6 +370,50 @@ private function initializeData(array $initialData, array $expectedData = null)
         }
     }
 
+    private function prepareBulkWriteArguments($arguments)
+    {
+        $operations = [];
+        $operations['options'] = $arguments['options'];
+        foreach ($arguments['requests'] as $requests) {
+            if (isset($requests['arguments']['arrayFilters'])) {
+                if (isset($requests['arguments']['options'])) {
+                    array_push($requests['arguments']['options'], ['arrayFilters' => $requests['arguments']['arrayFilters']]);
+                } else {
+                    $options = ['arrayFilters' => $requests['arguments']['arrayFilters']];
+                }
+            }
+            $innerArray = [];
+            switch ($requests['name']) {
+                case 'deleteMany':
+                case 'deleteOne':
+                    $innerArray = [$requests['arguments']['filter']];
+                    if (isset($requests['arguments']['options'])) {
+                         array_push($innerArray, $requests['arguments']['options']);
+                    }
+                case 'insertOne':
```

Comment:
The absence of any `break` statements here means that each `case` falls through to the next. That's likely what you want for `deleteMany` and `deleteOne` but not for truly unrelated operations. Otherwise, you're going to clobber the `$inneryArray` value by the end of this `switch` block.